a3p-integration/proposals/a:upgrade-next/exit-reclaim.test.js

```diff
@@ -0,0 +1,61 @@
+import test from 'ava';
+import { execFile } from 'node:child_process';
+import { promisify } from 'node:util';
+
+const offerId = 'bad-invitation-15'; // cf. prepare.sh
+const from = 'gov1';
+
+test.before(async t => {
+  const $ = async (file, args) => {
+    console.log('$', file, ...args);
+    const { stdout } = await promisify(execFile)(file, args);
+    return stdout.replace(/\n$/, '');
+  };
+  Object.assign(t.context, { $ });
+});
+
+test('exitOffer tool reclaims stuck payment', async t => {
+  const { $ } = t.context;
+
+  const addr = await $('agd', [
+    ...'keys --keyring-backend=test show -a'.split(' '),
+    from,
+  ]);
+  t.log(from, 'addr', addr);
+
+  const getBalance = async () => {
+    const txt = await $('agd', [
+      ...'query bank balances -o json'.split(' '),
+      addr,
+    ]);
+    const { balances } = JSON.parse(txt);
+    const { amount } = balances.find(({ denom }) => denom === 'uist');
+    return Number(amount);
+  };
+
+  const getBlock = async () => {
+    const txt = await $('agd', ['status']);
+    const status = JSON.parse(txt);
+    return Number(status?.SyncInfo?.latest_block_height);
+  };
+
+  const waitBlock = async (n = 1) => {
+    const start = await getBlock();
+    let block = start;
+    while (block < start + n) {
+      console.log('...block', block, '<', start + n);
+      await new Promise(r => setTimeout(r, 1000));
+      block = await getBlock();
+    }
+  };
+
+  const before = await getBalance();
+  t.log(from, { before });
+  await $('node', ['./exitOffer.js', '--id', offerId, '--from', from]);
```

Comment:
```suggestion
  // UNTIL https://github.com/Agoric/agoric-sdk/issues/9259
  await $('agops', ['inter', 'bid', 'cancel', offerId, '--from', from]);
```

Reply:
Do we know that `agops` is in `$PATH`? from the shell, it seems to be an alias. trying it out...

Reply:
Ok. We could change that throw to a warn but given that you already have `exitOffer.js` working I agree it's fine to proceed with it.

Reply:
Ok. We could change that throw to a warn but given that you already have `exitOffer.js` working I agree it's fine to proceed with it.